doc/releases/reef.rst

```diff
@@ -0,0 +1,219 @@
+  mechanism has been added: the ``ceph-objectstore-tool`` has added a
+  ``trim-pg-log-dups`` operation that targets situations where an OSD is unable
+  to boot due to inflated dups. When this is the case, the ``You can be hit by
+  THE DUPS BUG`` warning will be visible in the OSD logs. Relevant tracker:
+  https://tracker.ceph.com/issues/53729
+* The RADOS Python bindings are now able to process (opt-in) omap keys as bytes
+  objects. This enables interacting with RADOS omap keys that are not
+  decodeable as UTF-8 strings.
+* mClock Scheduler: The mClock scheduler (the default scheduler in Quincy) has
+  undergone significant usability and design improvements to address the slow
+  backfill issue. The following is a list of some important changes:
+
+  * The ``balanced`` profile is set as the default mClock profile because it
+    represents a compromise between prioritizing client I/O and prioritizing
+    recovery I/O. Users can then choose either the ``high_client_ops`` profile
+    to prioritize client I/O or the ``high_recovery_ops`` profile to prioritize
+    recovery I/O.
+  * QoS parameters including ``reservation`` and ``limit`` are now specified in 
+    terms of a fraction (range: 0.0 to 1.0) of the OSD's IOPS capacity.
+  * The cost parameters (``osd_mclock_cost_per_io_usec_*`` and
+    ``osd_mclock_cost_per_byte_usec_*``) have been removed. The cost of an 
+    operation is now a function of the random IOPS and maximum sequential
+    bandwidth capability of the OSD's underlying device.
+  * Degraded object recovery is given higher priority than misplaced
+    object recovery because degraded objects present a data safety issue that
+    is not present with objects that are merely misplaced. As a result,
+    backfilling operations with the ``balanced`` and ``high_client_ops`` mClock
+    profiles might progress more slowly than in the past, when backfilling
+    operations used the 'WeightedPriorityQueue' (WPQ) scheduler.
+  * The QoS allocations in all the mClock profiles are optimized in 
+    accordance with the above fixes and enhancements.
+  * For more details, see:
+    https://docs.ceph.com/en/latest/rados/configuration/mclock-config-ref/
+* The ``active_clients`` array displayed by the ``ceph mgr dump`` command now
+  has a ``name`` field that shows the name of the manager module that
+  registered a RADOS client. Previously, the ``active_clients`` array showed
+  the address of a module's RADOS client, but not the name of the module.
+
+RBD
+~~~
+
+* RBD: The semantics of compare-and-write C++ API (`Image::compare_and_write`
+  and `Image::aio_compare_and_write` methods) now match those of the C API.
+  Both the compare and the write steps operate only on ``len`` bytes, even if
+  their respective buffers are larger. The previous behavior that compared only
+  up to the size of the compare buffer was prone to subtle breakage when
+  straddling a stripe unit boundary.
```

Comment:
@zdover23 - Edited/reviewed version from 2023 June 18: RBD: The semantics of compare-and-write C++ API (``Image::compare_and_write`` and ``Image::aio_compare_and_write`` methods) now match those of C API. Both compare and write steps operate only on ``len`` bytes even if the buffers associated with them are larger. The previous behavior of comparing up to the size of the compare buffer was prone to subtle breakage upon straddling a stripe unit boundary.

Reply:
Accepted.